source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx

```diff
@@ -58,6 +60,36 @@ export const DocumentDetailForm: React.FunctionComponent<
     [props.documentTypes],
   );
 
+  const [documentTypePurpose, setDocumentTypePurpose] = useState(null);
+
+  const matchDocumentType = useCallback(
+    (documentTypeId: number) => {
+      const purpose = props.documentTypes.find(
```

Comment:
Hmm, you need to do a type conversion of documentTypeId even though it is already typed as a number?

Reply:
good catch, I think I reused the function from a place where the doctype was a string (the form)